README.md

```diff
@@ -8,6 +8,73 @@ The privML Privacy Evaluator is a tool that assesses ML model's levels of privac
+
+The current implementation gives the user the choice of the attack model classifier. The default model is a neural network, but a random forest and gradient boosting models can be chosen as well.
+
+Important to note is that this attack performs faster on specifically larger datasets than its brother, the Label-Only Membership Inference attack (see next attack), which in return offers robustness against privacy defense mechanisms that modify a model's output (confidence values) in order to thwart attacks.
+In general, this attack should be the preferred one, and in case of bad Precision and Recall values, the Label-Only-based membership inference attack should be applied.
+
+![](docs/mia_blackbox.png)
+
+#### Membership Inference Back-Box Label-Only Attack - Decision Boundary
+
+This is the implementation of the initial idea of a black-box membership inference Label-Only attack as provided by [Choquette-Choo et al.](https://arxiv.org/abs/2007.14321).
+
+The attack is also motivated by the protection of the privacy of the individuals whose data was used to train a model. Specifically, the attack provided here aims to investigate what data records were used to train a model, judging not by confidence values, but instead by using model outputs in the shape of labels only.
+
+This is a black-box membership attack which only requires a predicted label as the output of the target model and not a vector of probabilities. This fact makes the attack much more general and applicable to the cases where the attacker does not have access to the output probabilities of the black-box model.
+
+This attack performs on par with attacks that require access to model confidences as shown in its reference paper.
+
+If the above confidence-based membership inference attack seems to perform poorly, it might be advisable to use this very attack here, as so called confidence-masking mechanism might be existent.
+
+![](docs/mia_blackbox_decision_boundary.png)
+
+#### Membership Inference Black-Box Rule-Based Attack
+
+This attack is motivated by the protection of the privacy of the population from which a set of data is drawn from. Although the nature of a valid model is based on statistical facts about the population, hence making it difficult to protect against population privacy breaches, this attack aims to help demonstrate a model’s weakness in regards to population membership inference.
```

Comment:
The first sentence is redundant again, please remove. The second sentence is more introductory and general to all MIAs. Therefore, remove here. --> remove whole paragraph, maybe take information from second sentence somewhere to introduction above.